src/graph/immutable_graph.cc

```diff
@@ -0,0 +1,944 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/immutable_graph.cc
+ * \brief DGL immutable graph index implementation
+ */
+
+#include <stdlib.h>
+#include <dgl/immutable_graph.h>
+
+#ifdef _MSC_VER
+// TODO(zhengda) it seems MS compiler doesn't have rand_r. Let's
+// use rand for now. This may not have good performance because rand has
+// a global variable shared by all threads.
```

Comment:
Not the case for Windows; [VC Runtime is always multithreaded](https://social.msdn.microsoft.com/Forums/vstudio/en-US/3263ab0b-da3f-4737-9d80-0788a5b426db/randr-identifier-not-found?forum=vcgeneral).